### PR TITLE
python3Packages.url-normalize: init at 1.4.3

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
URL normalization for Python

https://github.com/niksite/url-normalize

This is a Home Assistant dependency.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotlambda** [comment]
@fabaff Why update `component-packages.nix` if both components using `url-normalize` are still missing something else?

File: pkgs/development/python-modules/url-normalize/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+, pytestCheckHook
+, six
+}:
+
+buildPythonPackage rec {
+  pname = "url-normalize";
+  version = "1.4.3";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
```
Comment (dotlambda) [inline pkgs/development/python-modules/url-normalize/default.nix:18]: Why not `fetchPypi`? I case that applies, you should add a comment that tests are missing.

File: pkgs/development/python-modules/url-normalize/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    pytest-socket
+    pytestCheckHook
+  ];
+
+  pythonImportsCheck = [ "url_normalize" ];
+
+  meta = with lib; {
+    description = "URL normalization for Python";
+    homepage = "https://github.com/niksite/url-normalize";
+    license = with licenses; [ mit ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/url-normalize/default.nix:42]:
```suggestion
    license = licenses.mit;
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 109994` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python37Packages.url-normalize</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>python38Packages.url-normalize</li>
  </ul>
</details>

File: pkgs/development/python-modules/url-normalize/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/url-normalize/default.nix:18]: Not a lot of packages have that and I wouldn't nit people about it. So it is nice to have but not hard required.

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109994` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>python37Packages.url-normalize</li>
<li>python38Packages.url-normalize</li>
</ul>
</details>

**Comment by fabaff** [comment]
> @fabaff Why update component-packages.nix if both components using url-normalize are still missing something else?

Because `component-packages.nix` is updated every time `parse-requirements.py` is run. If we don't update `component-packages.nix` when a new dependency is available then we end up with conflicts. I tried to de-couple the update of `component-packages.nix` from the dependency addition (mainly to reduce the amount of commits) but the next person had to update the file, `parse-requirements.py` picked-up both dependencies and this is not an ideal situation if stuff has to be reverted.
